functions/nodejs-apps.bash

```diff
@@ -202,3 +202,147 @@ nodered_setup() {
     dashboard_add_tile "nodered"
   fi
 }
+
+## Function for downloading zigbee2mqtt to current system
+##
+##    zigbee2mqtt_download(String prefix)
+##
+zigbee2mqtt_download() {
+  echo -n "$(timestamp) [openHABian] Downloading zigbee2mqtt... "
+  #if ! [[ -d "${1}/zigbee2mqtt" ]]; then
+    cond_echo "\\nFresh Installation... "
+    if ! cond_redirect mkdir /opt/zigbee2mqtt; then echo "FAILED (mkdir /opt/zigbee2mqtt)";echo1; fi
+    if ! cond_redirect chown openhabian /opt/zigbee2mqtt; then echo "FAILED (chown /opt/zigbee2mqtt)";echo 2; fi
+    if ! cond_redirect chgrp openhab /opt/zigbee2mqtt; then echo "FAILED (chgrp /opt/zigbee2mqtt)";echo 3; fi
+    if ! cond_redirect sudo -u "${username:-openhabian}" git clone https://github.com/Koenkk/zigbee2mqtt.git "/opt/zigbee2mqtt"; then echo "FAILED (git clone)"; return 1; fi
+  #fi
+}
+
+## Function for installing zigbee2mqtt.
+##
+##    zigbee2mqtt_setup()
+##
+zigbee2mqtt_setup() {
+  local zigbee2mqttBase
+  local z2mInstalledText="A configuration for Zigbee2MQTT is already existing.\\n\\nWould you like to update Zigbee2MQTT to the latest version with this configuration?"
+  local introText="A MQTT-server is required for zigbee2mqtt. If you haven't installed one yet, please select <cancel> and come back after installing one (e.g. Mosquitto).\\n\\nZigbee2MQTT will be installed from the official repository.\\n\\nDuration is about 4 minutes..."
+  local installText="Zigbee2MQTT is installed from the official repository.\\n\\nPlease wait about 4 minutes..."
+  local uninstallText="Zigbee2MQTT is completely removed from the system."
+  local adapterText="Please select your zigbee adapter:"
+  local mqttUserText="\\nPlease enter your MQTT-User (default = openhabian):"
+  local mqttPWText="\\nIf your MQTT-server requires a password, please enter it here:"
+  local adapterArray=()
+  local my_adapters=""
+  local mqttDefaultUser="openhabian"
+  local mqttUser
+  local serverIP; serverIP="$(hostname -I)"; serverIP=${serverIP::-1} # remove trailing space
+  local installSuccessText="Setup was successful. Zigbee2MQTT is now up and running.\\n\\nFor further Zigbee-settings open frontend (in 2 minutes): \\nhttp://${serverIP}:8081.\n\nDocumentation of ZigBee2MQTT:\\nhttps://www.zigbee2mqtt.io/guide/configuration"
+  local updateSuccessText="Update successful. \\n\\nFor further Zigbee-settings open frontend (in 2 minutes): \\nhttp://${serverIP}:8081.\n\nDocumentation of Zigbee2MQTT:\\nhttps://www.zigbee2mqtt.io/guide/configuration"
+  local loopSel=1
+  local i=0
+
+  if [[ $1 == "remove" ]]; then
+    if [[ -n $INTERACTIVE ]]; then
+      if ! (whiptail --title "Zigbee2MQTT Uninstall" --yes-button "Continue" --no-button "Cancel" --yesno "$uninstallText" 7 80); then echo "CANCELED"; return 0; fi
+    fi
+    echo -n "$(timestamp) [openHABian] Removing Zigbee2MQTT service ... "
```

Comment:
Remove the space after the end of the word and the start of the ellipsis to stay consistent with the rest of the codebase.
```suggestion
    echo -n "$(timestamp) [openHABian] Removing Zigbee2MQTT service... "
```